script/test/cmd/tests.go

```diff
@@ -0,0 +1,127 @@
+	} else {
+		fmt.Println("Test doesn't Pass")
+	}
+}
+
+func ExpectSuccessOpenShift(dockerComposeFile, result string) {
+	b, _ := ioutil.ReadFile(result)
+	output := string(b)
+	cmdName := "kompose"
+	cmdArgs := []string{"--provider=openshift", "convert", "-f", dockerComposeFile, "--stdout", "-j"}
+	cmd, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error Cmd", err)
+		os.Exit(1)
+	}
+	if reflect.DeepEqual(output, string(cmd)) {
+		fmt.Println("Test Pass")
+	} else {
+		fmt.Println("Test doesn't Pass")
+	}
+}
+
+func ExpectSuccessAndWarningK8s(dockerComposeFile, result string) {
+	warning := "WARN"
+	info := "INFO"
+	b, _ := ioutil.ReadFile(result)
+	output := string(b)
+	cmdName := "kompose"
+	cmdArgs := []string{"convert", "--stdout", "-f", dockerComposeFile, "-j"}
+	cmd, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error Cmd", err)
+		os.Exit(1)
+	}
+	file := string(cmd)
+	if strings.Contains(file, warning) || strings.Contains(file, info) {
+		str, err := exec.Command(cmdName, cmdArgs...).Output()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "Err Cmd", err)
+			os.Exit(1)
+		}
+		if reflect.DeepEqual(output, string(str)) {
+			fmt.Println("Test Pass")
+		} else {
+			fmt.Println("Test doesn't Pass")
+		}
+	}
+}
+
+func ExpectSuccessAndWarningOpenShift(dockerComposeFile, result string) {
+	warning := "WARN"
+	info := "INFO"
+	b, _ := ioutil.ReadFile(result)
+	output := string(b)
+	cmdName := "kompose"
+	cmdArgs := []string{"--provider=openshift", "convert", "--stdout", "-f", dockerComposeFile, "-j"}
+	cmd, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error Cmd", err)
+		os.Exit(1)
+	}
+	file := string(cmd)
+	if strings.Contains(file, warning) || strings.Contains(file, info) {
+		str, err := exec.Command(cmdName, cmdArgs...).Output()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "Err Cmd", err)
+			os.Exit(1)
+		}
+		if reflect.DeepEqual(output, string(str)) {
+			fmt.Println("Test Pass")
+		} else {
+			fmt.Println("Test doesn't Pass")
+		}
+	}
+}
+
+func ExpectFailureK8s(dockerComposeFile string) {
+	cmdName := "kompose"
+	cmdArgs := []string{"convert", "--stdout", "-f", dockerComposeFile, "-j"}
+	_, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error Cmd")
+		fmt.Println("Test Pass")
+	}
+}
+
+func ExpectFailureOpenShift(dockerComposeFile string) {
+	cmdName := "kompose"
+	cmdArgs := []string{"--provider=openshift", "convert", "--stdout", "-f", dockerComposeFile, "-j"}
+	_, err := exec.Command(cmdName, cmdArgs...).CombinedOutput()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "Error Cmd")
+		fmt.Println("Test Pass")
+	}
+}
+
+func main() {
+	ExpectSuccessK8s()
+	ExpectSuccessOpenShift()
+	ExpectSuccessAndWarningK8s()
+	ExpectSuccessAndWarningOpenShift()
+	ExpectFailureK8s()
+	ExpectFailureOpenShift()
+}
```

Comment:
It would also be great if you could do it using the golang subtests way, please read about it, and you could also use this as a reference - https://github.com/redhat-developer/opencompose/blob/master/pkg/encoding/v1/encoding_test.go